x/profiles/types/genesis.go

```diff
@@ -31,12 +37,41 @@ func ValidateGenesis(data *GenesisState) error {
 		return err
 	}
 
-	for _, transferReq := range data.DtagTransferRequests {
-		err := transferReq.Validate()
+	for _, req := range data.DtagTransferRequests {
+		if !profileExists(data.Profiles, req.Sender) {
+			return fmt.Errorf("invalid DTag transfer request; sender does not exist: %s", req.Sender)
+		}
+
+		if !profileExists(data.Profiles, req.Receiver) {
+			return fmt.Errorf("invalid DTag transfer request; receiver does not exist: %s", req.Receiver)
+		}
+
+		err := req.Validate()
 		if err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
+
+// containDuplicates tells whether or not the profiles slice contain duplicates of the given profile
+func containDuplicates(profiles []Profile, profile Profile) bool {
+	var count = 0
+	for _, p := range profiles {
+		if p.Equal(profile) {
+			count++
```

Comment:
Wasn't it better to return `true` here so it stops when found a duplicate and not go through all the profiles array?
Am I missing some purpose with these implementation?

Reply:
@bragaz This method is used to make sure that the given `profile` does not appear 2 or more times inside the profiled slice. This is used to check that, for any profile, there are no duplicates of that profile. This means that inside the given slice, there will always be 1 instance of that profile, but we want to check if there are 2 or more. For this reason, we cannot stop at the first instance that is equal to the given profile, because that will be the profile itself. Instead, we need to count all the duplicated occurrences and return whether or not there are 2 or more.

Reply:
Oh right. Gotcha